components/aws/sagemaker/common/_utils.py

```diff
@@ -221,5 +226,174 @@ def print_tranformation_job_result(output_location):
+    request['TrainingJobDefinition']['StoppingCondition']['MaxRuntimeInSeconds'] = args['max_run_time']
+    request['TrainingJobDefinition']['EnableNetworkIsolation'] = args['network_isolation']
+    request['TrainingJobDefinition']['EnableInterContainerTrafficEncryption'] = args['traffic_encryption']
+    request['TrainingJobDefinition']['RoleArn'] = args['role']
+
+    ### Update or pop warm start configs
+    if args['warm_start_type'] and args['parent_hpo_jobs']:
+        request['WarmStartConfig']['WarmStartType'] = args['warm_start_type']
+        parent_jobs = [n.strip() for n in args['parent_hpo_jobs'].split(',')]
+        for i in range(len(parent_jobs)):
+            request['WarmStartConfig']['ParentHyperParameterTuningJobs'].append({'HyperParameterTuningJobName': parent_jobs[i]})
+    else:
+        if args['warm_start_type'] or args['parent_hpo_jobs']:
+            if not args['warm_start_type']:
+                logging.error('Must specify warm start type as either "IdenticalDataAndAlgorithm" or "TransferLearning".')
+            if not args['parent_hpo_jobs']:
+                logging.error("Must specify at least one parent hyperparameter tuning job")
+            # TODO: determine what should actually happen here
+            # raise Exception('Could not create job request')
+            logging.info("Proceeding without warm start")
+        request.pop('WarmStartConfig')
+
+    ### Update tags
+    for key, val in args['tags'].items():
+        request['Tags'].append({'Key': key, 'Value': val})
+
+    return request
+
+
+def create_hyperparameter_tuning_job(client, args):
+    """Create a Sagemaker HPO job"""
+    request = create_hyperparameter_tuning_job_request(args)
+    try:
+        job_arn = client.create_hyper_parameter_tuning_job(**request)
+        hpo_job_name = request['HyperParameterTuningJobName']
+        logging.info("Created Hyperparameter Training Job with name: " + hpo_job_name)
+        logging.info("URL: https://{}.console.aws.amazon.com/sagemaker/home?region={}#/hyper-tuning-jobs/{}".format(args['region'], args['region'], hpo_job_name))
+        return hpo_job_name
+    except ClientError as e:
+        raise Exception(e.response['Error']['Message'])
+
+
+def wait_for_hyperparameter_training_job(client, hpo_job_name):
+    ### Wait until the job finishes
+    while(True):
+        response = client.describe_hyper_parameter_tuning_job(HyperParameterTuningJobName=hpo_job_name)
+        status = response['HyperParameterTuningJobStatus']
+        if status == 'Completed':
+            logging.info("Hyperparameter tuning job ended with status: " + status)
+            break
+        if status == 'Failed':
+            message = response['FailureReason']
+            logging.error('Hyperparameter tuning failed with the following error: {}'.format(message))
+            raise Exception('Hyperparameter tuning job failed')
+        logging.info("Hyperparameter tuning job is still in status: " + status)
```

Comment:
if the job has stopped then it will go in infinite loop. 
we should check for progress case and continue the loop for failure case we should just return.

Reply:
Looks like `Completed` and `Failed` are only two end state? Job will eventually fit into these two? @carolynwang Can you confirm?

Reply:
Yes that's right. Eventually if/when stopping the job when the run is terminated is implemented, the job can also be Stopping/Stopped state but the pod+container+script should be terminated anyway